components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -111,20 +96,40 @@ impl QueuedTaskConversion for DedicatedWorkerScriptMsg {
             WorkerScriptMsg::Common(script_msg) => script_msg,
             _ => return None,
         };
-        let (category, boxed, pipeline_id) = match script_msg {
-            CommonScriptMsg::Task(category, boxed, pipeline_id) =>
-                (category, boxed, pipeline_id),
+        let (category, boxed, pipeline_id, task_source) = match script_msg {
+            CommonScriptMsg::Task(category, boxed, pipeline_id, task_source) =>
+                (category, boxed, pipeline_id, task_source),
             _ => return None,
         };
-        Some((Some(worker), category, boxed, pipeline_id))
+        Some((Some(worker), category, boxed, pipeline_id, task_source))
     }
 
     fn from_queued_task(queued_task: QueuedTask) -> Self {
-        let (worker, category, boxed, pipeline_id) = queued_task;
-        let script_msg = CommonScriptMsg::Task(category, boxed, pipeline_id);
+        let (worker, category, boxed, pipeline_id, task_source) = queued_task;
+        let script_msg = CommonScriptMsg::Task(
+            category,
+            boxed,
+            pipeline_id,
+            task_source
+        );
         DedicatedWorkerScriptMsg::CommonWorker(worker.unwrap(), WorkerScriptMsg::Common(script_msg))
     }
 
+    fn task_source_name(&self) -> Option<&TaskSourceName> {
+        let common_worker_msg = match self {
+            DedicatedWorkerScriptMsg::CommonWorker(_, common_worker_msg) => common_worker_msg,
+            _ => return None,
+        };
+        let script_msg = match common_worker_msg {
+            WorkerScriptMsg::Common(ref script_msg) => script_msg,
+            _ => return None,
+        };
+        match script_msg {
+            CommonScriptMsg::Task(_category, _boxed, _pipeline_id, source_name) => Some(&source_name),
+            _ => return None,
+        }
+    }
+
```

Comment:
nit: If you insert the new code(in this case `task_source_name`) at the same level as the old code(in this case `task_category`), it makes the diff easier to understand for the reviewer(the diff will also show more clearly the difference between the old and new code, since the entire block wasn't changed in fact, see the diff for the other implementations).

In this case, since all other implementations have `task_source_name` as the first method, as does the trait signature, there is also a benefit in terms of consistency at keeping it the first(top-most) method for this implementation as well.